src/basic/cgroup-util.c

```diff
@@ -160,6 +160,17 @@ bool cg_freezer_supported(void) {
         return supported;
 }
 
+bool cg_kill_supported(void) {
+        static thread_local int supported = -1;
+
+        if (supported >= 0)
+                return supported;
+
+        supported = cg_all_unified() > 0 && access("/sys/fs/cgroup/init.scope/cgroup.kill", F_OK) == 0;
```

Comment:
hmm, for new code I'd prefer if we'd debug log about unexpected errors, i.e. cg_ns_supported() does this nicely. cg_freezer_supported() does not (which i figure you used as blueprint).

i.e. we'd expect that access() either succeeds or returns ENOENT. Any other error indicates something is off.